hypershift-operator/ignitionserver/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+// This is the simplest http server that enable us to satisfy
+// 1 - 1 relation between clusters and ign endpoints.
+// At the moment this acts as purely a proxy towards MCS that are created on demand by NodePools,
+// this might / might not change in the future.
+// TODO (alberto): Support only https.
+// TODO (alberto): Support only token authenticated requests.
+// TODO (alberto): Parameterize listening port and URL to proxy.
+// TODO (alberto): Metrics.
+
+type ignHandler struct{}
+
+var _ http.Handler = &ignHandler{}
+
+func (h *ignHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// We only match /ign/$NodePoolName
+	ignPathPattern, err := regexp.Compile("^/ign/[^/ ]*$")
+	if err != nil {
+		panic(err)
+	}
+
+	log.Printf("User Agent %s", r.Header.Get("User-Agent"))
+	log.Printf("Requested %s", r.URL.Path)
+
+	if ignPathPattern.MatchString(r.URL.Path) {
+		urlPathSplit := strings.SplitAfter(r.URL.Path, "/")
+		// This can only happen if our regExp is not working as expected.
+		if len(urlPathSplit) > 3 {
+			panic(fmt.Sprintf("Unexpected URL path: %s", r.URL.Path))
```

Comment:
Maybe log and return an http response (400?) instead of crashing?

Reply:
This is not expected by design, If it ever happen I'd consider it a fatal bug therefore why I went for panic.
Happy to go either way.